### PR TITLE
Fax Responders

### PR DESCRIPTION
**Pull request by realforest2001**
# About the pull request

This project is intended to update and overhaul the fax machines, in regards to who can contact who and how. I have implemented some of these changes within this PR already, but more is to come.
It also adds a dedicated role for responding to faxes, split into subsidiary jobs for each main faction (USCM/USCM Provost/WY/TWE/UPP/CLF/CMB)
This role is experimental, and will either be a dedicated staff role, or potentially opened up to players. The method of this opening however is still to be determined, whether it will act like a traditional whitelist, or be operated differently is as of yet unknown.
There is a wiki page for the fax responder role that can be found here: [https://cm-ss13.com/wiki/Fax_Responder](url).

# Explain why it's good for the game
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
add: Added unique identifier to all fax machines.
add: Added a new way to contact fax machines directly, titled Specific Machine Code in the select department setting.
add: Added a Fax Responder role, broken down into subsidiary jobs for each faction. This is currently experimental in handling, and access is restricted.
add: Fax Machines will tell you (when selecting a faction's High Command) if a Fax Responder is awake.
code: Makes the Whitelist Panel use the correct permissions for Overseer.
/:cl:

### CODE REVIEW

File: code/modules/gear_presets/fax_responders.dm
Comment (Drulikar) [inline code/modules/gear_presets/fax_responders.dm:29]:
```suggestion
```
Removed in https://github.com/cmss13-devs/cmss13/pull/7205

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by Nanu308 — APPROVED** [review]
neat map, lgtm

**Comment by Rem1-code** [comment]
USCM HC fax responder is locked out of the remote monitoring computer, and doesn't seem to have access to any paperwork prefabs.

**Comment by realforest2001** [comment]
> New setup is **great** and gives a lot of awareness to better prepare faxes ahead of time. Small bugs for your consideration, monitoring camera:
> 
> * Planetary spawnzones sometimes are completely frozen and you can't move. 'riding' the DS down circumvents this.
> * You collide with objects(thankfully, they do not collide with you.) - so you cannot move past mobs but they can move past you. ...Unfortunately you do collide with walls and doors - which means you can open doors up like a ghost.
> 
> Happened on Fiorina and LV-624 respectively once each, so didn't get to test reproducability

I've not noticed any collisions, please let me know if you replicate this.
As for not being able to move the remote cameras, you can't move them unless they are or will be within 5 tiles of a friendly camera. This means no spying on remote parts of the colony unless a camera friendly to you is already there.

**Comment by MPhonks** [comment]
I can imagine something of a smaller alert similar to when CIC staff or SEA latejoins for the role in common chat made by ARES, "Fax Tubes for USCM Provost have been recalibrated." If you send a fax before then, you could get a warning that it may not be answered. With a fax operator, you can be pretty sure it'll get adressed. Very much a QOL for any roleplaying since there'd be confirmation that what you're writing isn't going to be a timewaste.

File: code/__DEFINES/mode.dm
Comment (realforest2001) [inline code/__DEFINES/mode.dm]: Remind me to put FACTION_FAX back into this before merge, it's removed for TM compat with hunting grounds.

File: code/__DEFINES/mode.dm
Comment (realforest2001) [inline code/__DEFINES/mode.dm:296]:
```suggestion
#define FACTION_LIST_HUMANOID list(FACTION_MARINE, FACTION_PMC, FACTION_WY, FACTION_WY_DEATHSQUAD, FACTION_CLF, FACTION_CONTRACTOR, FACTION_MARSHAL, FACTION_UPP, FACTION_FREELANCER, FACTION_SURVIVOR, FACTION_NEUTRAL, FACTION_COLONIST, FACTION_MERCENARY, FACTION_DUTCH, FACTION_HEFA, FACTION_GLADIATOR, FACTION_PIRATE, FACTION_PIZZA, FACTION_SOUTO, FACTION_YAUTJA, FACTION_ZOMBIE, FACTION_TWE, FACTION_FAX)
```

File: code/modules/client/client_procs.dm
Comment (harryob) [inline code/modules/client/client_procs.dm:894]:
```suggestion
```
🙂

File: code/game/gamemodes/cm_initialize.dm
Comment (harryob) [inline code/game/gamemodes/cm_initialize.dm:261]:
```suggestion
			//FAX RESPONDER INITIALIZE\\
```
🤓

File: code/game/gamemodes/cm_initialize.dm
Comment (harryob) [inline code/game/gamemodes/cm_initialize.dm:324]:
```suggestion
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Pull request merged**
